feat: bootstrap @invisible/release package.

rodrigo4244 opened the pull request.

rodrigo4244 commented on src/release.js:
```diff
@@ -0,0 +1,66 @@
+  }
+
+  const client = github.client(GITHUB_TOKEN)
+  const repo = currentRepo()
```
It is not actually the current repo. It is `invisible-tech/release`. If you do not update your `origin` remote, you are going to have problems.

rodrigo4244 replied:
This is the best idea I could come with. Does this work on circleCI? Hmmmm to verify.

rodrigo4244 commented on src/release.js:
```diff
@@ -0,0 +1,66 @@
+  }
+
+  const notes = lastChangelogUpdate({ changelogFile: 'ANNOUNCEMENTS.md' }) || ''
+  if (isEmpty(notes)) {
```
Should we skip when there's no release note? Maybe don't

rodrigo4244 commented:
Need to add `GITHUB_TOKEN` to circleCI.
Make it work from the command line? How the consumer is supposed to add the `GITHUB_TOKEN`?

rodrigo4244 commented:
To add README.md

rodrigo4244 commented on src/helpers.js:
```diff
@@ -0,0 +1,36 @@
+  return trim(branch)
+}
+
+const currentRepo = () => {
```
This command actually works on Circle CI, it seems like circleCI set the origin url on their side :)

rodrigo4244 commented on src/helpers.js:
```diff
@@ -0,0 +1,36 @@
+
+const currentRepo = () => {
+  const { stdout: remotes } = spawn.sync(
+    'git',
```
This actually work on circle CI, it seems like they set origin url on their end :)

rodrigo4244 commented:
The circle error is because `assert-version-bump` and `assert-changelog-update` does not work on the first PR.

rodrigo4244 commented:
This is the release I created from the command line on https://github.com/invisible-tech/merge-parsers/pull/25
![image](https://user-images.githubusercontent.com/8367325/32630812-2ca0016c-c585-11e7-985d-88ca330cdb40.png)

rodrigo4244 commented on README.md:
````diff
@@ -1,20 +1,72 @@
+
+```javascript
+  'use strict'
+  const { release } = require('@invisible/release')
````
Should I export the function since it is only the release one? Should I "merge" this package inside `@invisible/publish`?

keenahn replied:
yes just export the one function unless you think we're going to add more to this package later

rodrigo4244 commented on bin/release:
```diff
@@ -0,0 +1,46 @@
+const release = require('../src/release')
+
+try {
+  const dotenv = require('dotenv')
```
If we use programmatically, `dotenv` is not loaded, `GITHUB_TOKEN` is required. Do you agree with me?

rodrigo4244 commented on bin/release:
```diff
@@ -0,0 +1,47 @@
+  })
+  .version()
+  .help()
+  .wrap(null)
```
This is just to fit the whole terminal width.

rodrigo4244 commented:
Should we get optionally `changelogFile` from `package.json` or it would be feature creep?

keenahn commented on README.md:
```diff
@@ -1,20 +1,72 @@
+
+  // draft is set to false by default.
+  // changelogFile is set to ANNOUNCEMENTS.md by default
+  release({ GITHUB_TOKEN, draft: true, changelogFile: 'ANNOUNCEMENTS.md' })
```
why is the default `ANNOUNCEMENTS.md` ? The default should be `CHANGELOG.md`

rodrigo4244 replied:
Agreed!

keenahn commented on src/helpers.js:
```diff
@@ -0,0 +1,36 @@
+  return trim(branch)
+}
+
+const currentRepo = () => {
```
these should be in their own package since we reuse them a lot :)

rodrigo4244 replied:
Card added to Trello -> https://trello.com/c/T6KyBBzN/499-convert-invisible-release-src-helpersjs-into-its-own-package

keenahn commented on src/release.js:
```diff
@@ -0,0 +1,61 @@
+
+const pkg = finder().next().value || {}
+
+const run = async ({ GITHUB_TOKEN, draft = false, changelogFile = 'ANNOUNCEMENTS.md' } = {}) => {
```
just call it `githubToken` 

and default changelogFile should be `CHANGELOG.md`

rodrigo4244 replied:
Agreed!

keenahn commented on src/release.js:
```diff
@@ -0,0 +1,61 @@
+  return {
+    pass: true,
+    msg: 'new release created',
+  }
```
why? Why not just throw on errors

rodrigo4244 replied:
To let the user handle how he wants programmatically the errors, but, on the binary we exit with code 1.

rodrigo4244 commented:
I changed the default to be `CHANGELOG.md`, renamed `GITHUB_TOKEN` to `githubToken` inside the `src/release.js` and exported a `function` instead of an `object`. Did the necessary changes on `README.md` and `circle.yml`.

A good way to test this package is:
1. Pull this branch
2. Modify `src/release.js` line 26 to be `  if (currentBranch() !== 'rgo/bootstrap') {`
3. Run `bin/release -d -c=ANNOUNCEMENTS.md`
4. Go to https://github.com/invisible-tech/release/releases and see that a draft has been created.
5. Discard the draft

Thanks for the good feedback! :100:

keenahn commented:
@gunar delegating this to you, but you can delegate back to me if you think this is a priority.

gunar commented:
Confirming I own this.

gunar commented:
@kewitz passing this back to you because I don't know if we want to continue this. You be the judge.

rodrigo4244 commented:
Legacy
